assets/js/modules/analytics/datastore/properties.js

```diff
@@ -0,0 +1,325 @@
+const FETCH_PROPERTIES_PROFILES = 'FETCH_PROPERTIES_PROFILES';
+const RECEIVE_CREATE_PROPERTY = 'RECEIVE_CREATE_PROPERTY';
+const RECEIVE_CREATE_PROPERTY_FAILED = 'RECEIVE_CREATE_PROPERTY_FAILED';
+const RECEIVE_PROPERTIES_PROFILES = 'RECEIVE_PROPERTIES_PROFILES';
+const RECEIVE_PROPERTIES_PROFILES_FAILED = 'RECEIVE_PROPERTIES_PROFILES_FAILED';
+
+export const INITIAL_STATE = {
+	isDoingCreateProperty: {},
+	isFetchingPropertiesProfiles: {},
+	properties: undefined,
+};
+
+export const actions = {
+	/**
+	 * Creates a new Analytics property.
+	 *
+	 * Creates a new Analytics property for an existing Google Analytics account.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Object} accountId Google Analytics account ID.
+	 * @return {Object} Redux-style action.
+	 */
+	*createProperty( accountId ) {
+		invariant( accountId, 'accountId is required.' );
+
+		try {
+			const response = yield actions.fetchCreateProperty( accountId );
+
+			const { property } = response;
+
+			yield actions.receiveCreateProperty( { accountId, property } );
+			yield Data.dispatch( STORE_NAME ).setPropertyID( property.id );
+			return;
+		} catch ( error ) {
+			// TODO: Implement an error handler store or some kind of centralized
+			// place for error dispatch...
+			return actions.receiveCreatePropertyFailed( { accountId, error } );
+		}
+	},
+
+	*fetchCreateProperty( accountId ) {
+		invariant( accountId, 'accountId is required.' );
+
+		return {
+			payload: { accountId },
+			type: FETCH_CREATE_PROPERTY,
+		};
+	},
+
+	*fetchPropertiesProfiles( accountId ) {
+		invariant( accountId, 'accountId is required.' );
+
+		return {
+			payload: { accountId },
+			type: FETCH_PROPERTIES_PROFILES,
+		};
+	},
+
+	/**
+	 * Adds a property to the data store.
+	 *
+	 * Adds the newly-created property to the existing properties in
+	 * the data store.
+	 *
+	 * @since n.e.x.t
+	 * @private
+	 *
+	 * @param {Object} args Argument params.
+	 * @param {string} args.accountId Google Analytics account ID.
+	 * @param {Object} args.property Google Analytics property object.
+	 * @return {Object} Redux-style action.
+	 */
+	receiveCreateProperty( { accountId, property } ) {
+		invariant( accountId, 'accountId is required' );
+		invariant( property, 'property is required' );
+
+		return {
+			payload: { accountId, property },
+			type: RECEIVE_CREATE_PROPERTY,
+		};
+	},
+
+	/**
+	 * Logs an error with property creation.
+	 *
+	 * @since n.e.x.t
+	 * @private
+	 *
+	 * @param {Object} args Argument params.
+	 * @param {string} args.accountId Google Analytics account ID.
+	 * @param {Object} args.error Error object.
+	 * @return {Object} Redux-style action.
+	 */
+	receiveCreatePropertyFailed( { accountId, error } ) {
+		invariant( accountId, 'accountId is required' );
+		invariant( error, 'error is required.' );
+
+		return {
+			payload: { accountId, error },
+			type: RECEIVE_CREATE_PROPERTY_FAILED,
+		};
+	},
+
+	receivePropertiesProfiles( { accountId, properties, profiles } ) {
```

Comment:
This also needs to support the _optional_ `matchedProperty` in the same way `accounts.js` needs to.